Meta-Test: Race condition in testBackup share links

Error Line: https://github.com/SJSU-CS-systems-group/wildstore/blob/7de1b5ed226ddb9bd6a9601374a844fce38ac291/wildstore-meta/src/test/java/edu/sjsu/wildstore/meta/MetaTest.java#L75

Recreate error run: `mvn package`

Output:
`
[ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.517 s <<< FAILURE! -- in edu.sjsu.wildstore.meta.MetaTest
[ERROR] edu.sjsu.wildstore.meta.MetaTest.testBackup -- Time elapsed: 0.003 s <<< ERROR!
java.nio.file.NoSuchFileException: /var/folders/hn/1n55qvmj6g5_cdf31mwk2cgc0000gn/T/junit-1640751945480916550/share-links.json
        at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
        at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:57)
        at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:161)
        at java.base/java.nio.file.Files.readAttributes(Files.java:1702)
        at java.base/java.nio.file.Files.getLastModifiedTime(Files.java:2184)
        at edu.sjsu.wildstore.meta.MetaTest.testBackup(MetaTest.java:75)
        at java.base/java.lang.reflect.Method.invoke(Method.java:565)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)
`

--- a/wildstore-meta/src/test/java/edu/sjsu/wildstore/meta/MetaTest.java
+++ b/wildstore-meta/src/test/java/edu/sjsu/wildstore/meta/MetaTest.java
@@ -72,11 +72,20 @@ public class MetaTest {
 
     @Test
     public void testBackup() throws IOException, InterruptedException {
-        FileTime firstTime = Files.getLastModifiedTime(backupDir.resolve("share-links.json"));
-        Assertions.assertTrue(Files.exists(backupDir.resolve("share-links.json")));
-        Assertions.assertTrue(Files.exists(backupDir.resolve("userData.json")));
+        Path shareLinks = backupDir.resolve("share-links.json");
+        Path userData = backupDir.resolve("userData.json");
 
+        // Wait up to 3s for the first backup run to create the files
+        long deadline = System.currentTimeMillis() + 3000;
+        while ((!Files.exists(shareLinks) || !Files.exists(userData)) && System.currentTimeMillis() < deadline) {
+            Thread.sleep(100);
+        }
+
+        Assertions.assertTrue(Files.exists(shareLinks));
+        Assertions.assertTrue(Files.exists(userData));
+
+        FileTime firstTime = Files.getLastModifiedTime(shareLinks);
         Thread.sleep(1000);
-        Assertions.assertTrue(firstTime.compareTo(Files.getLastModifiedTime(backupDir.resolve("share-links.json"))) < 0);
+        Assertions.assertTrue(firstTime.compareTo(Files.getLastModifiedTime(shareLinks)) < 0);
     }
 }

